doc/node-types-html/index.html.in

```diff
@@ -15,7 +15,229 @@
+	            var hidden_code = $('<textarea></textarea>');
+	            hidden_code.appendTo('#example-info');
+	            hidden_code.attr('id','example-info-code');
+	            hidden_code.disabled=true;
+	            /* adding the get-code button */
+	            var getCodeButton = $('<img></img>');
+	            getCodeButton.attr('src','images/button_get_code.png');
+	            getCodeButton.attr('id','example-get-code');
+	            info.append(getCodeButton);
+	            var code = $('#example-image').data('code'); 
+	            $.get(String(code),function(data){              //Getting code from path
+	                hidden_code.html(String(data));
+	            }); 
+	            getCodeButton.on('click',copyExampleCode);
+	            }
+	        }
+
+	        /* handles the click on a thumbnail */
+	        function onThumbExampleClick(event){
+	            loadExampleImage($(this));
+	        }
+
+	        /* loads the correspondent image from the thumbnail */
+	        function loadExampleImage(_thumb)
+	        {
+	            /* removing old image reference */
+	            $('#example-image-container').empty();
+	            $("img[id^=thumb_]").each(function(){
+	                $(this).removeClass('example-thumb-active');
+	            });
+
+	            /* loading selected image */
+	            var imageURL = $(_thumb).data('large');
+
+	            // SVG Loading happens here 
+	            var image = $("<img></img");
+	            image.attr("src",imageURL);
+	            image.attr('id','example-image');
+	            image.data('code',$(_thumb).data('code'));
+
+	            /* adding the loaded image into the drag view */
+	            $("#example-image-container").append(image);
+
+	            /* marking the active thumb */
+	            $(_thumb).addClass('example-thumb-active');
+
+	            /* making the image draggable */
+	            image.draggable();
+
+	            /* removing existing copy feedback and restoring the copy button */
+	            if($('#example-info').has('#example-code-copied')){
+	                $('#example-code-copied').remove();
+	                $('#example-get-code').show();
+	            }
+	        }
+
+	        /* --------------------------------------------------------
+	         * COPY EXAMPLE CODE
+	         * Implement the algorithm to read the example
+	         * code here. The var code already stores url 
+	         * when creating the Entry in the main loop
+	         * --------------------------------------------------------
+	         */
+	        function copyExampleCode(event){
+	            var x = location.protocol;
+	            var y = /^file?:/i;
+	            if(y.test(x)){
+	                alert("Error: Code cannot be copied! \nPlease run Cheat Sheet from server");
+	            }
+	            else{
+	                var text = document.getElementById('example-info-code');
+	                selecttext(text);
+	                copycode();
```

Comment:
This function copycode returns FALSE if the copy fails, you don't need to test if cheat-sheet is being run from 
"file://" to show an error. 

Like you've done here

```
if(y.test(x)){
     alert("Error: Code cannot be copied! \nPlease run Cheat Sheet from server");
}
```

What if is there a browser that does not block the copy code when running from local ? We would show a fake error to the user, right ?

Why not testing if copy code returns false and throw an generic error for the user ?

I would do something like this:

```
if (!copycode()) {
   alert("Failed to copy code");
}
```

You can also get that was generated from the try catch inside the the copycode to show more details of the error.

This would be better way better.